diff -crB dealii-9.3.1/include/deal.II/base/thread_local_storage.h dealii-9.3.1-patched/include/deal.II/base/thread_local_storage.h
*** dealii-9.3.1/include/deal.II/base/thread_local_storage.h	Wed Jun 30 16:01:32 2021
--- dealii-9.3.1-patched/include/deal.II/base/thread_local_storage.h	Wed Jul  7 19:25:48 2021
***************
*** 250,255 ****
--- 250,258 ----
       */
  #  ifdef DEAL_II_HAVE_CXX17
      mutable std::shared_mutex insertion_mutex;
+ #  elif __APPLE__
+     #include <boost/thread/shared_mutex.hpp>
+     mutable boost::shared_mutex insertion_mutex;
  #  else
      mutable std::shared_timed_mutex insertion_mutex;
  #  endif
